ci/run_ci.sh

```diff
@@ -0,0 +1,69 @@
+
+#!/bin/bash
+
+source ci/env.sh
+
+test_choice=$@
+
+if [[ ${test_choice} != '' ]]; then
+  echo -e "Running chosen tests: "
+  for i in ${test_choice};do
+    echo -e "${i}"
+  done
+  test_list=${test_choice}
+else
+  echo "Running full test"
+  test_list=`cat ci/CI_test_list.yml`           #yml file contains the list of test scripts to run from different workloads
+  echo -e  "$test_list\n\n"                         
+fi
+
+diff_list=`git diff --name-only origin/master`
+echo -e "List of files changed : ${diff_list} \n"
+
+cat > results.markdown << EOF
+Results for e2e-benchmarking CI Tests
+Workload                | Test                           | Result | Runtime  |
+------------------------|--------------------------------|--------|----------|
+EOF
+
+
+test_rc=0
+
+for test in ${test_list}; do            
+
+  start_time=`date`
+
+  command=${test##*:}                            #to extract the shell script name to run
+  directory=${test%:*}                           #to extract the workload directory name 
+  echo $directory  $command
+
+  figlet "CI test for ${test}"                   # figlet gives a heading for each test
+
+  cd workloads/
+  cd $directory
+  echo $PWD
+
+  (sleep 900s; sudo pkill -f $command) &      #to kill a workload script if it doesn't execute within 500s ; runs in the background
```

Comment:
We might want to make this configurable with a default to make sure we cover the test cases which might take longer than the specified duration in the future? Thoughts?

Reply:
yes makes sense , will add an env var named TEST_TIMEOUT in env.sh which can be initialized with a default value of 10 or 15 mins .